pandas/_testing.py

```diff
@@ -2550,6 +2551,7 @@ def assert_produces_warning(
     filter_level="always",
     check_stacklevel=True,
     raise_on_extra_warnings=True,
+    match=None,
```

Comment:
can you annoate

Reply:
Annotated, but presumably the code became slightly less pythonic. Sorry for that.